hibernate-core/src/test/java/org/hibernate/event/service/internal/EventListenerReplacementStrategyTest.java

```diff
@@ -0,0 +1,298 @@
+package org.hibernate.event.service.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.hibernate.event.service.spi.DuplicationStrategy;
+import org.hibernate.event.service.spi.EventListenerGroup;
+import org.hibernate.event.service.spi.EventListenerRegistrationException;
+import org.hibernate.event.spi.ClearEvent;
+import org.hibernate.event.spi.ClearEventListener;
+import org.hibernate.event.spi.EventType;
+
+import org.hibernate.testing.TestForIssue;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test that a listener replacing the original one is actually called when the event is fired for each listener.
+ * <p>
+ *     Note: I'm using ClearEvent for the tests because it's the simpler one I've found.
+ * </p>
+ */
+@TestForIssue(jiraKey = "HHH-13831")
+public class EventListenerReplacementStrategyTest {
```

Comment:
Can we rename this `EventListenerDuplicationStrategyTest`?  It now tests more than just the replace strategy

Reply:
Done!